include/seqan3/alignment/matrix/detail/score_matrix_single_column.hpp

```diff
@@ -0,0 +1,246 @@
+ * This score matrix scores a column for the optimal score and the horizontal score and uses a virtual column that
+ * stores only one value of type `score_t` for the vertical column. Hence, this matrix can only be used for a column
+ * based computation layout.
+ *
+ * ### Range interface
+ *
+ * The matrix offers a input range interface over the columns of the matrix. Dereferencing the iterator will return
+ * another range which represents the actual score column in memory. The returned range is a
+ * transformed seqan3::views::zip view over the optimal, horizontal and vertical column. The reference type of this
+ * view is the seqan3::detail::affine_cell_proxy, which offers a practical interface to access the value of the
+ * optimal, horizontal and vertical value of the underlying matrices.
+ */
+template <typename score_t>
+class score_matrix_single_column
+{
+private:
+    //!\brief The type of the score column which allocates memory for the entire column.
+    using physical_column_t = std::vector<score_t>;
+    //!\brief The type of the virtual score column which only stores one value.
+    using virtual_column_t = decltype(views::repeat_n(score_t{}, 1));
+
+    class matrix_iterator;
+
+    //!\brief The column over the optimal scores.
+    physical_column_t m_optimal_column{};
+    //!\brief The column over the horizontal gap scores.
+    physical_column_t m_horizontal_column{};
+    //!\brief The virtual column over the vertical gap scores.
+    virtual_column_t m_vertical_column{};
+    //!\brief The number of columns for this matrix.
+    size_t m_number_of_columns{};
+
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    score_matrix_single_column() = default; //!< Defaulted.
+    score_matrix_single_column(score_matrix_single_column const &) = default; //!< Defaulted.
+    score_matrix_single_column(score_matrix_single_column &&) = default; //!< Defaulted.
+    score_matrix_single_column & operator=(score_matrix_single_column const &) = default; //!< Defaulted.
+    score_matrix_single_column & operator=(score_matrix_single_column &&) = default; //!< Defaulted.
+    ~score_matrix_single_column() = default; //!< Defaulted.
+    //!\}
+
+    /*!\brief Resets the matrix and reallocates the memory of the underlying matrix.
+     * \tparam sequence1_t The type of the first sequence; must model std::ranges::forward_range.
+     * \tparam sequence2_t The type of the second sequence; must model std::ranges::forward_range.
+     *
+     * \param[in] sequence1 The first sequence.
+     * \param[in] sequence2 The second sequence.
+     */
+    template <std::ranges::forward_range sequence1_t, std::ranges::forward_range sequence2_t>
+    void reset_matrix(sequence1_t && sequence1, sequence2_t && sequence2)
+    {
+        size_t number_of_rows = std::ranges::distance(sequence2) + 1;
+        m_optimal_column.resize(number_of_rows, 0);
+        m_horizontal_column.resize(number_of_rows, 0);
```

Comment:
Either the function has a bad name and/or brief, or the `resize` here is wrong, or assumes something that is not written in the docs.

If optimal/horizontal column contains values, resizing them won't reset the vector.
```
std::vector<int> vec{1,2,3,4};
vec.resize(6, 0); // 1,2,3,4,0,0
```

And if I am pedantic: This function may or may not reallocate the memory depending on the capacity?

Reply:
ok, yes it might be a little confusing about what should happen. I updated the documentation accordingly.